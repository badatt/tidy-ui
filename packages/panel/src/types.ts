import { HTMLAttributes } from 'react';
import { IBaseProps } from '@tidy-ui/types';

/**
 * PanelContext props
 */
export interface IPanelContext {
  /**
   * If `true`, sets the panel expanded
   *
   * @default false
   */
  expanded?: boolean;

  /**
   * Toggle expand/close of the panel
   */
  toggle: () => void;
}

/**
 * Panel props
 */
export interface IPanelProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * If `true`, set the panel expanded
   *
   * @default false
   */
  expanded?: boolean;
  /**
   * Margin, css `margin` equivalent
   */
  margin?: string;
}

/**
 * PanelHeader props
 */
export interface IPanelHeaderProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Element that encloses the header
   */
  ele?: JSX.Element;
}

/**
 * PanelBody props
 */
export interface IPanelBodyProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Element that encloses the body
   */
  ele?: JSX.Element;
  /**
   * If `true`, set the body visible
   *
   * @default false
   */
  visible?: boolean;
}

/**
 * PanelGroup props
 */
export interface IPanelGroupProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Margin, css `margin` equivalent
   */
  margin?: string;
}
