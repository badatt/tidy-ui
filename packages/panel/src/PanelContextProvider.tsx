import React from 'react';
import { IPanelContext, IPanelProps } from './types';

const PanelContext = React.createContext<IPanelContext | null>(null);

/**
 * PanelContextProvided to share state across all panel related components
 *
 * @param {IPanelProps} props Panel props
 * @returns {JSX.Element} react node
 */
const PanelContextProvider = (props: IPanelProps): JSX.Element => {
  const { children } = props;
  const [isExpanded, setIsExpanded] = React.useState(props.isExpanded);

  const toggle = React.useCallback(() => {
    setIsExpanded(!isExpanded);
  }, [isExpanded]);

  const panelContextValue = React.useMemo(() => ({ isExpanded, toggle }), [isExpanded, toggle]);

  return <PanelContext.Provider value={panelContextValue}>{children}</PanelContext.Provider>;
};

export { PanelContext, PanelContextProvider };
