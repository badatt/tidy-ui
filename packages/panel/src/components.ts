import { css, hsla, styled } from '@tidy-ui/commons';
import { IPanelBodyProps, IPanelProps } from './types';

const PanelRoot = styled.div<IPanelProps>`
  display: flex;
  flex-direction: column;
  width: 100%;
  ${({ theme: { layout, palette, isDark } }) => css`
    border: 1px solid ${isDark ? palette.neutral[700] : palette.neutral[400]};
    border-radius: ${layout.radius};
    background-color: ${isDark ? hsla(palette.neutral.shades[900], 0.3) : palette.neutral[50]};
  `}
`;

const PanelHeaderRoot = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  vertical-align: middle;
  cursor: pointer;
  padding: 1rem 1rem;
`;

const Icon = styled.i<IPanelProps>`
  display: flex;
  justify-content: center;
  height: 1rem;
  width: 1rem;
  ${({ theme: { palette, isDark }, expanded }) => css`
    transform: ${expanded ? 'rotate(180deg)' : 'rotate(0deg)'};
    color: ${isDark ? palette.neutral[500] : palette.neutral[700]};
  `}
  transition: all 200ms cubic-bezier(0.075, 0.82, 0.165, 1);
`;

const PanelBodyRoot = styled.div<IPanelBodyProps>`
  padding: 0rem 1rem;
  transition: all 200ms cubic-bezier(0.075, 0.82, 0.165, 1);
  visibility: visible;
  ${({ visible }) => css`
    visibility: ${visible ? 'visible' : 'hidden'};
    opacity: ${visible ? '1' : '0'};
    height: ${visible ? 'fit-content' : '0'};
    padding-bottom: ${visible ? '1rem' : '0'};
  `}
`;

const PanelGroupRoot = styled.div<IPanelProps>`
  ${({ theme: { layout, palette, isDark } }) => css`
    border: 1px solid ${isDark ? palette.neutral[700] : palette.neutral[400]};
    border-radius: ${layout.radius};
    & > * {
      border: 0;
    }

    & > :not(:last-child)::after {
      content: '';
      border-bottom: 0.5px solid ${isDark ? palette.neutral[700] : palette.neutral[400]};
    }
  `}
`;

export { Icon, PanelBodyRoot, PanelGroupRoot, PanelHeaderRoot, PanelRoot };
