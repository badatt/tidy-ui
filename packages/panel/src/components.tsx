import { applyStandardOverrideStyles, color, createFontStyle, css, hsla, styled } from '@tidy-ui/commons';
import { IPanelBodyProps, IPanelGroupProps, IPanelProps } from './types';

const PanelRoot = styled.div<IPanelProps>`
  display: flex;
  flex-direction: column;
  ${applyStandardOverrideStyles}
`;

/** @internal */
interface IActionIconProps {
  /** @internal */
  isExpanded?: boolean;
}

const ActionIcon = styled.span<IActionIconProps>`
  min-width: 1.5rem;
  height: 1.5rem;
  width: 1.5rem;
  background-color: ${hsla(color.slate[400], 0.3)};
  border-radius: 50%;
  transition: all 200ms cubic-bezier(0.075, 0.82, 0.165, 1);
  ${({ isExpanded }) =>
    css`
      transform: ${isExpanded ? 'rotate(180deg)' : 'rotate(0deg)'};
    `}
  &:hover {
    background-color: ${hsla(color.slate[500], 0.4)};
  }
`;

const PanelHeaderRoot = styled.div`
  display: flex;
  align-items: center;
  gap: 0.5rem;
  vertical-align: middle;
  cursor: pointer;
  margin: 0.5rem 0;
  ${applyStandardOverrideStyles}
`;

const PanelBodyRoot = styled.div<IPanelBodyProps>`
  display: block;
  //padding: 0rem 1rem;
  transition: all 200ms cubic-bezier(0.075, 0.82, 0.165, 1);
  visibility: visible;
  ${createFontStyle()}
  ${({ isVisible, height }) => css`
    visibility: ${isVisible ? 'visible' : 'hidden'};
    opacity: ${isVisible ? '1' : '0'};
    height: ${isVisible ? 'fit-content' : '0'};
    padding-bottom: ${isVisible ? '1rem' : '0'};
    ${isVisible
      ? css`
          height: ${height ?? 'fit-content'};
        `
      : css`
          height: 0;
        `}
    ${height &&
    css`
      overflow-y: auto;
    `}
  `}
  ${applyStandardOverrideStyles}
`;

const PanelGroupRoot = styled.div<IPanelGroupProps>`
  ${({ theme: { isDark }, margin, hasSeparator }) => css`
    margin: ${margin};
    ${hasSeparator &&
    css`
      & > :not(:last-child)::after {
        content: '';
        border-bottom: 1px solid ${isDark ? hsla(color.slate[600]) : hsla(color.slate[400])};
      }
    `}
  `}
  ${applyStandardOverrideStyles}
`;

export { ActionIcon, PanelBodyRoot, PanelGroupRoot, PanelHeaderRoot, PanelRoot };
