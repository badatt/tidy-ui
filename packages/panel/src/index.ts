export * from './Panel';
export * from './PanelBody';
export * from './PanelGroup';
export * from './PanelHeader';
