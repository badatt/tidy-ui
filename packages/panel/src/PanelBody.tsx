import React from 'react';
import { PanelBodyRoot } from './components';
import { PanelContext } from './PanelContextProvider';
import { IPanelBodyProps, IPanelContext } from './types';

const PanelBody = React.forwardRef<HTMLDivElement, IPanelBodyProps>((props, ref) => {
  const { children, ...rest } = props;
  const { expanded } = React.useContext(PanelContext) as IPanelContext;
  return (
    <PanelBodyRoot ref={ref} role="presentation" {...rest} visible={expanded}>
      {children}
    </PanelBodyRoot>
  );
});

PanelBody.defaultProps = {
  visible: false,
};

export { PanelBody };
