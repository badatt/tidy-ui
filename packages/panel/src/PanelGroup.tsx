import React from 'react';
import { PanelGroupRoot } from './components';
import { IPanelGroupProps } from './types';

const PanelGroup = React.forwardRef<HTMLDivElement, IPanelGroupProps>((props, ref) => {
  const { children, ...rest } = props;
  return (
    <PanelGroupRoot role="article" ref={ref} {...rest}>
      {children}
    </PanelGroupRoot>
  );
});

PanelGroup.defaultProps = {
  hasSeparator: false,
};

PanelGroup.displayName = 'PanelGroup';

export { PanelGroup };
