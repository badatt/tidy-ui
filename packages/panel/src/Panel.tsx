import React from 'react';
import { PanelRoot } from './components';
import { PanelBody } from './PanelBody';
import { PanelContextProvider } from './PanelContextProvider';
import { PanelHeader } from './PanelHeader';
import { IPanelProps } from './types';

/** @internal */
interface PanelComponent extends React.ForwardRefExoticComponent<IPanelProps & React.RefAttributes<HTMLDivElement>> {
  /** @internal */
  Body: typeof PanelBody;
  /** @internal */
  Header: typeof PanelHeader;
}

/**
 * Panel can be used to fold data, show only necessary data and can expand whenever needed by the user. Panel brings accordion effect
 */
const Panel = React.forwardRef<HTMLDivElement, IPanelProps>((props, ref) => {
  const { children, ...rest } = props;
  return (
    <PanelContextProvider {...props}>
      <PanelRoot ref={ref} role="article" {...rest}>
        {children}
      </PanelRoot>
    </PanelContextProvider>
  );
}) as PanelComponent;

Panel.defaultProps = {
  expanded: false,
};

Panel.Body = PanelBody;
Panel.Header = PanelHeader;

export { Panel };
