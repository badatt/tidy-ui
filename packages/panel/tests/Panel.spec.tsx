/**
 * @jest-environment jsdom
 */
import React from 'react';
import { act } from 'react-dom/test-utils';
import { cleanup, fireEvent, getByRole, queryByRole, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight, TidyUiProvider } from '../../commons/src';
import { Tone } from '../../types/src';
import { Panel } from '../src';

afterEach(cleanup);

const text = `Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique, maxime. Aliquam, ea neque? Quibusdam
          itaque quos earum! Ex, neque, unde officia accusamus necessitatibus, quas incidunt architecto a impedit ut
          ullam. Veritatis saepe placeat ipsum aliquid laudantium quod, reprehenderit recusandae quas mollitia pariatur?
          Eos dolorum numquam voluptatum sapiente, accusantium voluptate animi odio tempora totam reprehenderit adipisci
          nulla praesentium exercitationem architecto quaerat. Explicabo quis dolores incidunt. Animi obcaecati quis
          pariatur minus, non praesentium. Voluptatum magnam laborum nemo molestias amet architecto, maiores blanditiis
          omnis sunt quae similique aliquid delectus est animi eligendi voluptate? Nulla voluptatum excepturi, tempora
          voluptates cum dolore facere, ducimus officiis ut dolores vero necessitatibus numquam aliquam sapiente ab
          eligendi, alias provident beatae tempore iure itaque possimus. Ullam dignissimos explicabo eum. Provident
          architecto ipsum, nobis explicabo repudiandae quaerat nemo deleniti asperiores ex voluptatibus reprehenderit
          cumque fugit non quos doloribus reiciendis nihil. Omnis cupiditate vel iure placeat cumque repellat
          necessitatibus voluptate qui!`;

describe('Panel', () => {
  it('Basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel>
          <Panel.Header>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Panel.Header>
          <Panel.Body>{text}</Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Dark mode basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <Panel>
          <Panel.Header>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Panel.Header>
          <Panel.Body>{text}</Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Visible', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel>
          <Panel.Header>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Panel.Header>
          <Panel.Body isVisible>{text}</Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Toggle visible', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel>
          <Panel.Header isToggleVisible>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Panel.Header>
          <Panel.Body isVisible>{text}</Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Visible with height', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel>
          <Panel.Header>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Panel.Header>
          <Panel.Body isVisible height="5rem">
            {text}
          </Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Accent', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        {Object.values(Tone)
          .filter((i) => !isNaN(Number(i)))
          .map((v, i) => (
            <Panel key={i}>
              <Panel.Header accent={Tone[v]}>
                Lorem ipsum dolor sit amet consectetur, adipisicing elit. Voluptas odit minima facere quo, quidem
                officiis et. Optio maiores eius deserunt, perspiciatis magnam tenetur nemo assumenda non commodi fugiat?
                Animi, necessitatibus!
              </Panel.Header>
              <Panel.Body accent={Tone[v]}>{text}</Panel.Body>
            </Panel>
          ))}
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Custom dimension', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel width="30rem">
          <Panel.Header>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Panel.Header>
          <Panel.Body height="25rem">{text}</Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Custom header element', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel>
          <Panel.Header ele={<div style={{ fontWeight: 'bold' }} />}>
            Lorem ipsum dolor sit amet consectetur adipisicing elit.
          </Panel.Header>
          <Panel.Body>{text}</Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Custom body element', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel isExpanded>
          <Panel.Header>Lorem ipsum dolor sit amet consectetur adipisicing elit.</Panel.Header>
          <Panel.Body ele={<div style={{ display: 'flex' }} />}>
            <div>{text}</div>
            <div>{text}</div>
            <div>{text}</div>
          </Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Toggle expand panel', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel>
          <Panel.Header accent="major" isToggleVisible>
            Lorem ipsum dolor sit amet consectetur adipisicing elit.
          </Panel.Header>
          <Panel.Body accent="major">{text}</Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    const { container } = tree;

    act(() => {
      userEvent.hover(container);
      const panelIcon = getByRole(container, 'button');
      expect(queryByRole(container, 'presentation')).toBeNull();
      expect(panelIcon).toBeVisible();
      fireEvent.click(panelIcon);
    });
    expect(queryByRole(container, 'presentation')).toBeVisible();
  });

  it('Toggle expand panel with custom height', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Panel>
          <Panel.Header accent="major" isToggleVisible>
            Lorem ipsum dolor sit amet consectetur adipisicing elit.
          </Panel.Header>
          <Panel.Body accent="major" height="250px">
            {text}
          </Panel.Body>
        </Panel>
      </TidyUiProvider>,
    );
    const { container } = tree;

    act(() => {
      userEvent.hover(container);
      const panelIcon = getByRole(container, 'button');
      expect(queryByRole(container, 'presentation')).toBeNull();
      expect(panelIcon).toBeVisible();
      fireEvent.click(panelIcon);
    });
    expect(queryByRole(container, 'presentation')).toBeVisible();
  });
});
