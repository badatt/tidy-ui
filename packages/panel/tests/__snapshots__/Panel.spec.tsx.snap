// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  border: 1px solid hsla(0,0%,64%,1);
  border-radius: 0.5rem;
  background-color: hsla(0,0%,98%,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  cursor: pointer;
  padding: 1rem 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1rem;
  width: 1rem;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: hsla(0,0%,25%,1);
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
}

.c3 {
  padding: 0rem 1rem;
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  visibility: visible;
  visibility: hidden;
  opacity: 0;
  height: 0;
  padding-bottom: 0;
}

<body>
    <div>
      <div
        class="c0"
        role="panel"
      >
        <div
          class="c1"
          role="panel-header"
        >
          <div
            classname="sc-dkzDqf kkXvAq"
            is="h6"
            role="text"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit
          </div>
          <i
            class="c2"
            role="panel-icon"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </i>
        </div>
        <div
          class="c3"
          role="panel-body"
        >
          <div
            classname="sc-dkzDqf bqmGt"
            is="p"
            role="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam ullam amet pariatur ducimus exercitationem quos rem vel nulla nostrum soluta? Repellat sit quam vero! Recusandae placeat adipisci quo qui odit!
          </div>
          <div
            classname="sc-dkzDqf bqmGt"
            is="p"
            role="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur quasi explicabo id amet sed labore, ex cumque repellendus inventore quas commodi at quis aspernatur libero voluptate maiores alias ipsum ducimus!
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  border: 1px solid hsla(0,0%,64%,1);
  border-radius: 0.5rem;
  background-color: hsla(0,0%,98%,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  cursor: pointer;
  padding: 1rem 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1rem;
  width: 1rem;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: hsla(0,0%,25%,1);
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
}

.c3 {
  padding: 0rem 1rem;
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  visibility: visible;
  visibility: hidden;
  opacity: 0;
  height: 0;
  padding-bottom: 0;
}

<div>
    <div
      class="c0"
      role="panel"
    >
      <div
        class="c1"
        role="panel-header"
      >
        <div
          classname="sc-dkzDqf kkXvAq"
          is="h6"
          role="text"
        >
          Lorem ipsum dolor sit, amet consectetur adipisicing elit
        </div>
        <i
          class="c2"
          role="panel-icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </i>
      </div>
      <div
        class="c3"
        role="panel-body"
      >
        <div
          classname="sc-dkzDqf bqmGt"
          is="p"
          role="text"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam ullam amet pariatur ducimus exercitationem quos rem vel nulla nostrum soluta? Repellat sit quam vero! Recusandae placeat adipisci quo qui odit!
        </div>
        <div
          classname="sc-dkzDqf bqmGt"
          is="p"
          role="text"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur quasi explicabo id amet sed labore, ex cumque repellendus inventore quas commodi at quis aspernatur libero voluptate maiores alias ipsum ducimus!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Panel Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  border: 1px solid hsla(0,0%,25%,1);
  border-radius: 0.5rem;
  background-color: hsla(0,0%,9%,0.3);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  cursor: pointer;
  padding: 1rem 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1rem;
  width: 1rem;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  color: hsla(0,0%,45%,1);
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
}

.c3 {
  padding: 0rem 1rem;
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  visibility: visible;
  visibility: visible;
  opacity: 1;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding-bottom: 1rem;
}

<body>
    <div>
      <div
        class="c0"
        role="panel"
      >
        <div
          class="c1"
          role="panel-header"
        >
          <div
            classname="sc-dkzDqf gQuYnE"
            is="h6"
            role="text"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit
          </div>
          <i
            class="c2"
            role="panel-icon"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </i>
        </div>
        <div
          class="c3"
          role="panel-body"
        >
          <div
            classname="sc-dkzDqf cbaKPf"
            is="p"
            role="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam ullam amet pariatur ducimus exercitationem quos rem vel nulla nostrum soluta? Repellat sit quam vero! Recusandae placeat adipisci quo qui odit!
          </div>
          <div
            classname="sc-dkzDqf cbaKPf"
            is="p"
            role="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur quasi explicabo id amet sed labore, ex cumque repellendus inventore quas commodi at quis aspernatur libero voluptate maiores alias ipsum ducimus!
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  border: 1px solid hsla(0,0%,25%,1);
  border-radius: 0.5rem;
  background-color: hsla(0,0%,9%,0.3);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  cursor: pointer;
  padding: 1rem 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 1rem;
  width: 1rem;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  color: hsla(0,0%,45%,1);
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
}

.c3 {
  padding: 0rem 1rem;
  -webkit-transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  transition: all 200ms cubic-bezier(0.075,0.82,0.165,1);
  visibility: visible;
  visibility: visible;
  opacity: 1;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding-bottom: 1rem;
}

<div>
    <div
      class="c0"
      role="panel"
    >
      <div
        class="c1"
        role="panel-header"
      >
        <div
          classname="sc-dkzDqf gQuYnE"
          is="h6"
          role="text"
        >
          Lorem ipsum dolor sit, amet consectetur adipisicing elit
        </div>
        <i
          class="c2"
          role="panel-icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </i>
      </div>
      <div
        class="c3"
        role="panel-body"
      >
        <div
          classname="sc-dkzDqf cbaKPf"
          is="p"
          role="text"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam ullam amet pariatur ducimus exercitationem quos rem vel nulla nostrum soluta? Repellat sit quam vero! Recusandae placeat adipisci quo qui odit!
        </div>
        <div
          classname="sc-dkzDqf cbaKPf"
          is="p"
          role="text"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur quasi explicabo id amet sed labore, ex cumque repellendus inventore quas commodi at quis aspernatur libero voluptate maiores alias ipsum ducimus!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
